docs/core/extensions/snippets/logging/library-authors/NonDiExampleService.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Logging.LibraryAuthors;
+
+public class NonDiExampleService
+{
+    private readonly ILogger<NonDiExampleService> _logger;
+
+    public NonDiExampleService() =>
+        _logger = NullLoggerFactory.Instance.CreateLogger<NonDiExampleService>();
+
+    public void DoSomething()
+    {
+        if (_logger.IsEnabled(LogLevel.Information))
```

Comment:
Similar to the previous example I'd suggest we drop the IsEnabled() check and switch to source-generated logging.